pkg/controllers/nodeclaim/termination/controller.go

```diff
@@ -89,7 +89,7 @@ func (c *Controller) Finalize(ctx context.Context, nodeClaim *v1beta1.NodeClaim)
 		return reconcile.Result{}, nil
 	}
 	if nodeClaim.Status.ProviderID != "" {
-		if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
+		if err = c.cloudProvider.Delete(ctx, nodeClaim); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil || cloudprovider.IsRetryableError(err) {
```

Comment:
What if we just re-leverage the existing NodeClaimNotFound error that we already use? We could just return a NotFound error once the instance is terminated like we do already today with our Get() and List() operations.

Reply:
@jonathan-innis If the instance is terminated then we don't want to return an error. It's only when it is not terminated that we want to return some form of error that will re-enqueue the reconciliation. And can't use  NodeClaimNotFound error in that case because we ignore that error here if I understand correctly.

Reply:
> And can't use NodeClaimNotFound error in that case because we ignore that error here if I understand correctly

You could keep trying to Delete() until you got the NodeClaimNotFound error that would indicate that the instance has been correctly terminated. I discussed with @njtran offline. I think it's fine to add this additional error type but I was generally trying to avoid us adding additional error types into our cloudprovider unless they had a specific reason for being there. Also, the other thing to consider here is whether returning an error on a Delete() call that succeeded so that the client can know that is should keep retrying makes a lot of sense. 

These are my thoughts, but I'm fine to take in what you have here and we can see how all the error types pan out over time.